### ISSUE TITLE
out of memory issue

### ISSUE THREAD

**Issue by jjhesk** [martinheidegger/mapslice#15]
I have uploaded an image with spec:
13.62MB
original dimension: 15822 x 1897

![basemap-1469553899965](https://cloud.githubusercontent.com/assets/629338/17148865/5c40a1a4-539b-11e6-91d8-51016960628c.jpg)

The issue that is running out memory on each every time. With the heroku free dyno it gives me R14 error on the first try when the RAM exceed 500MB. Then I  pull the code down and run it on windows with 8GB ram 64bit. 

I use parallel limit of 5 which is the default setting and i found that will crash every time on the zoom level 6. I think the better design show give the gm to release the ram when starting the next zoomlevel. that will squeeze out some memory from the process. i can think of another way using serial async process.

**Comment by martinheidegger** [comment]
I am sorry that your computers run out of memory. I think it might be good to have different async limits bases on the size of the current zoom level:
zoom level 1 (biggest size): parallel tasks 1
zoom level 2: parallel tasks 4
zoom level 3: parallel tasks 8
zoom level 4: parallel tasks 16 ...
etc.

**Comment by jjhesk** [comment]
is that number can be calculated based on the memory available in the system? For example, if you have found out that the current system have 512MB, 1GB, 2GB, 3GB,..

**Comment by martinheidegger** [comment]
No, its a logical step: The image processing system needs to be able to process the biggest image. With every zoom step the image should become 1/4th of the size, which means that the memory should stay constant at the minimum required.
